.ci/atime/tests.R

```diff
@@ -25,15 +25,23 @@ for(retGrp_setup in retGrp_values){
   for(retGrp_expr in retGrp_values){
     test.name <- sprintf("forderv(retGrp=%s-%s) improved in #4386", retGrp_setup, retGrp_expr)
     extra.test.list[[test.name]] <- list(
-    setup = substitute({
+    setup = quote({
       options(datatable.forder.auto.index = TRUE)
       set.seed(1)
       dt <- data.table(index = sample(N), values = sample(N))
-      data.table:::forderv(dt, "index", retGrp = RETGRP) # Initial sort to create the index and initialize caching
-    }, list(RETGRP=str2lang(retGrp_setup))),
+      index.list <- list()
+      for(retGrp in retGrp_values){
+        data.table:::forderv(dt, "index", retGrp = eval(str2lang(retGrp)))
+        index.list[[retGrp]] <- attr(dt, "index")
+      }
+    }),
     expr = substitute({
-      data.table:::forderv(dt, "index", retGrp = RETGRP) # Reusing the index and computing group info.
-    }, list(RETGRP=str2lang(retGrp_expr))),
+      setattr(dt, "index", index.list[[retGrp_setup]]) 
+      data.table:::forderv(dt, "index", retGrp = retGrp_expr) # Reusing the index and computing group info.
+    }, list(
+      retGrp_setup=retGrp_setup,
+      retGrp_expr=str2lang(retGrp_expr)
```

Comment:
in this refactor I compute both retGrp=T and F in setup (using CRAN data.table), and store the "index" attribute in `index.list`

then in expr I set the "index" attribute to one of the two indices we computed in setup (using either retGrp=T or F).
and then we call forderv using a git version of data.table...
and I observe a different result than before --- Fast is always linear (for any combination of retGrp=T or F), never constant.

I don't understand what is happening, or which version of this test we should prefer (with or without setattr).
Can someone who knows forderv better please investigate/explain what is expected time complexity here?